test/server/http/admin_test.cc

```diff
@@ -196,6 +202,81 @@ TEST_P(AdminInstanceTest, HelpUsesFormForMutations) {
   EXPECT_NE(-1, response.search(stats_href.data(), stats_href.size(), 0));
 }
 
+TEST_P(AdminInstanceTest, Runtime) {
+  Http::HeaderMapImpl header_map;
+  Buffer::OwnedImpl response;
+
+  std::unordered_map<std::string, const Runtime::Snapshot::Entry> entries{
+      {"string_key", {"foo", {}}}, {"int_key", {"1", {1}}}, {"other_key", {"bar", {}}}};
+  Runtime::MockSnapshot snapshot;
+  Runtime::MockLoader loader;
+
+  EXPECT_CALL(snapshot, getAll()).WillRepeatedly(testing::ReturnRef(entries));
+  EXPECT_CALL(loader, snapshot()).WillRepeatedly(testing::ReturnPointee(&snapshot));
+  EXPECT_CALL(server_, runtime()).WillRepeatedly(testing::ReturnPointee(&loader));
+
+  EXPECT_EQ(Http::Code::OK, admin_.runCallback("/runtime", header_map, response));
+
+  const std::string std_format{"int_key: 1\nother_key: bar\nstring_key: foo\n"};
+  EXPECT_NE(-1, response.search(std_format.data(), std_format.size(), 0));
```

Comment:
nit: Collapse all of this into a single EXPECT_EQ without any local variable and use BufferToString. Same in bad format test.